music_assistant/server/providers/tidal/__init__.py

```diff
@@ -864,27 +875,32 @@ def _parse_playlist(self, playlist_obj: TidalPlaylist) -> Playlist:
         if picture := (playlist_obj.square_picture or playlist_obj.picture):
             picture_id = picture.replace("-", "/")
             image_url = f"{RESOURCES_URL}/{picture_id}/750x750.jpg"
-            playlist.metadata.images = [
-                MediaItemImage(
-                    type=ImageType.THUMB,
-                    path=image_url,
-                    provider=self.instance_id,
-                    remotely_accessible=True,
-                )
-            ]
+            playlist.metadata.images = UniqueList(
+                [
+                    MediaItemImage(
+                        type=ImageType.THUMB,
+                        path=image_url,
+                        provider=self.instance_id,
+                        remotely_accessible=True,
+                    )
+                ]
+            )
 
         return playlist
 
     async def _iter_items(
-        self, func: Awaitable | Callable, *args, **kwargs
-    ) -> AsyncGenerator[Any, None]:
+        self,
+        func: Callable[_P, list[_R]] | Callable[_P, Awaitable[list[_R]]],
+        *args: _P.args,
+        **kwargs: _P.kwargs,
+    ) -> AsyncGenerator[_R, None]:
         """Yield all items from a larger listing."""
         offset = 0
         while True:
             if asyncio.iscoroutinefunction(func):
-                chunk = await func(*args, **kwargs, offset=offset)
+                chunk = await func(*args, **kwargs, offset=offset)  # type: ignore[arg-type]
```

Comment:
`Concatenate` doesn't seem to be flexible enough to handle this case, and there doesn't seem to be a way to add a constraint to the ParamSpec to verify `offset` is actually available in the wrapped function.